### ISSUE TITLE
I'm working on leg (master VS v1 VS my-contrib)

### ISSUE THREAD

**Issue by tst2005** [keplerproject/leg#3]
Hello,

`leg` seems a dead project but ...
I'm trying to revive it by a lot of minor changes.

I found the master branch that seems older than the v1 tag.

@hishamhm,
* do you know if the master is really older than the v1 tag ?
* is the current master is equal to leg 0.1.2 like the README said ?

I just created a v1.0.x branch from the v1 tag to be able to contrib on it.

I planned to make some minor change on the v1 : like mixed indentation fix to produce a cleaner v1 (planned to be the v1.0.1)
after that I will make changes to drop `module()` use (to a 1.2.0)

Regards,

**Comment by hishamhm** [comment]
> leg seems a dead project but ...
> I'm trying to revive it by a lot of minor changes.

That is great! If you want to take over the project, you could start by sending a PR with your updates and from there on we can look at things such as commit access and transfering the ownership of the entry at luarocks.org

> do you know if the master is really older than the v1 tag ?

I'm not really aware of the state of the repo, sorry :\

> is the current master is equal to leg 0.1.2 like the README said ?

Not sure either, but you should be able to download 0.1.2 from luarocks.org (the src package is a zip file and should contain everything) and diff it against master:

https://luarocks.org/modules/luarocks/leg/0.1.2-1

**Comment by tst2005** [comment]
I investigate on the history.
Now I understand what happened.

The original author, `hanjos` made some mistakes about version numbering, and create a v1 tag to ignore.

He publish on luaforge.net :
* 0.1
* 0.1.1
* 0.1.2
these 3 releases are available [here](http://files.luaforge.net/releases/leg/leg).

In the git's master branch, we can see other changes, not publish on luaforge.net.
`hanjos` merged `src/scanner.lua` feature into the `src/parser.lua` and increase the parser.lua version from `0.1.2` to `0.2`, but missed some other files like the README.

At the beginning of the projet, just after the first commit `hanjos` made a small minor commit tagged as v1.

In conclusion,

I will drop the v1 tag that is near the 0.1 version (but before the 0.1.1), this tag just sucks.
I will try to find the good commit equal to the 3 releases (maybe also the 4th)
I will drop all my useless recent work done on the v1 branch.

* -> remove the v1 tag.
* 0.1.0 -> tag v0.1.0+archive
* 0.1.1 -> tag v0.1.1+archive
* 0.1.2 -> tag v0.1.2+archive
* 0.2.0 -> tag v0.2.0+archive
* after that some people like me made some changes
* I planned to tag the current master with a name greater than v1

**Comment by tst2005** [comment]
After a deep analysis...

I found the commit that reflect the 0.1.2 release.I tagged it as v0.1.2.
I didn't find other commit that reflect the other previous version (0.1 and 0.1.1). I will don't spend more time to find them.

I tried to tag the 0.2 version (as a v0.2.0 tag) but there is no proper rockspec file to install this 0.2 version with luarocks.
I think the v0.2.0 release tag should inclure the missing rockspec file,  I tagged as v0.2.0+pre1 and v0.2.0+pre2, before fixing the missing rockspec file and possible other version number mismatch.

* dcd25ff (the current HEAD)
* (some commits skipped)
* dbb9c3e (tag: **v0.2.0-pre2**, r15) added rockspecs to source tree
* 0b562ba (tag: **v0.2.0-pre1**, r14) scanner: removed scanner.lua and related files, putting its functionality in parser.lua; updated documentation and examples
* d947b36 (r13) metalua-ast.lua: debugging, extra comments.
* 0948db6 (r12) metalua-ast.lua: A bigger example.
* ad774c5 (tag: **v0.1.2**, r11) README: added a Testing section.
* 2c21f77 (r10) doc/*, src/*.lua: regenerated the docs, updating the version number.
* cfc8c50 (r9) README: added binary operators' gripe parser.lua: added something, but took it off after, so no net change
* f37aa69 (r8) Added index.html
* 865bec7 (r7) Added COPYRIGHT, README and release.
* b5dd71c (r6) Makefile: removed some pointless code, such as SYS_BIN, TIMESTAMP and the           buffer directory in the bundle rule; changed PROJ_NAME to NAME
* d67785c (r5) *.html, *.css, *.gif: Documentation updated
* efd2068 (r4) Added leg rule, improved bundle rule
* 54cfe8c (r3) Updated Makefile
* fa04184 (r2) examples/*.lua: updated comments, some stylistic code changes doc/*.html: some slight text changes
* 4f6478d (r1) Initial revision

**Comment by hishamhm** [comment]
@tst2005 you're doing a great job on this! I have Admin access for you to this repo. What is your username on luarocks.org, so I can hand over the `leg` package to you, so you can take over the project?

**Comment by tst2005** [comment]
@hishamhm thanks,
I'm not ready yet to release new rock version of leg.

My initial goal was just to run the leg testsuite with lulpeg (instead of the lpeg).
I probably forgot a little my initial goal ;-)

**Comment by hishamhm** [comment]
@tst2005 Do you still wish to maintain this project? We are decomissioning the https://github.com/keplerproject org at GitHub, so I want to know whether we should archive this repository or move it elsewhere.

**Comment by tst2005** [comment]
@hishamhm I'm not actively working on it since a while but I still dreams to get a good lua parser/lexer. Archiving will definitively kill my motivation. Then if it does not cost too much, move it ! thanks :)

**Comment by hishamhm** [comment]
We're only moving to Lunar Modules projects with active maintainers, sorry. I see two possible options:

* I can transfer the project to your GitHub account now, and then you decide if you want to carry on with it;
* Or I can archive it and keep it here. Then, if you ever want to revisit this project, feel free to fork the archive and/or drop me a line so I can unarchive and transfer it.

**Comment by tst2005** [comment]
@hishamhm, Ok I understand. leg is clearly not actively maintained. Please transfert on my account (tst2005).

@hishamhm what is the good place to discuss the inclusion of another projet to lunar modules ?

Thanks again for all your work done on all other keplerproject's / lunamodules projects !
